src/DurableTask.Netherite/StorageLayer/Faster/FasterKV.cs

```diff
@@ -503,7 +503,8 @@ public override async Task<long> RunCompactionAsync(long target)
                     target - this.Log.BeginAddress,
                     this.GetElapsedCompactionMilliseconds());
 
-                var timeoutTask = Task.Delay(TimeSpan.FromMinutes(10));
+                var tokenSource = new CancellationTokenSource();
```

Comment:
`CancellationTokenSource` is an IDisposable and needs to be disposed at some point (otherwise you may be leaking resources such as kernel-provided synchronization primitives). See suggestion below on how to do that safely.

Reply:
I wish VS would warn me about these. I'm sure it can, if I had the right linting rules. I'll take your suggestion, thanks